### DISCUSSION TITLE
Adding Thai Model

### DISCUSSION THREAD

**Discussion by fws1** [idiap/coqui-ai-TTS#70] [Q&A] [answered]
Hi All,

Thank you for continuing to support the Coqui TTS code. I found this discussion group and hope you may be able to help me with adding an existing Thai model. There is no Thai model included that I can see. I am new to text-to-speech. I would like to convert Thai text to WAV files.

I can build and run the Docker image, and run the server to list and run models. Works fine from the command line (container and host) and interactively with a browser.

My issue is with trying to train the existing Thai model. I am following various READMEs with instructions from the old version with the new version 1.1 files.

Here are the references to the new Thai model and instructions:
https://github.com/wannaphong/KhanomTan-TTS-v1.1
https://huggingface.co/wannaphong/khanomtan-tts-v1.1
https://github.com/wannaphong/KhanomTan-TTS-v1.0/blob/main/train/README.md

What I'm trying is dropping the files "config_se.json" and "model_se.pth" into recipes/multilingual/vits_tts and executing "python3 recipes/multilingual/vits_tts/train_vits_tts.py". This results in the below output and error:

root@b1212a492ed7:~# python3 recipes/multilingual/vits_tts/train_vits_tts.py 
 > Training Environment:
 | > Backend: Torch
 | > Mixed precision: False
 | > Precision: float32
 | > Current device: 0
 | > Num. of GPUs: 1
 | > Num. of CPUs: 8
 | > Num. of Torch Threads: 4
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
 > Start Tensorboard: tensorboard --logdir=/root/recipes/multilingual/vits_tts/vits_vctk-August-13-2024_05+08AM-0000000

 > Model has 82244225 parameters

 > EPOCH: 0/1000
 --> /root/recipes/multilingual/vits_tts/vits_vctk-August-13-2024_05+08AM-0000000
 ! Run is kept in /root/recipes/multilingual/vits_tts/vits_vctk-August-13-2024_05+08AM-0000000
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1605, in fit
    self._fit()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1557, in _fit
    self.train_epoch()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1252, in train_epoch
    self.train_loader = self.get_train_dataloader(
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 725, in get_train_dataloader
    return self._get_loader(
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 689, in _get_loader
    loader = model.get_data_loader(
  File "/root/TTS/tts/models/vits.py", line 1624, in get_data_loader
    dataset.preprocess_samples()
  File "/root/TTS/tts/datasets/dataset.py", line 401, in preprocess_samples
    raise RuntimeError(msg)
RuntimeError: No samples left.

Am I on the right track here?

Any assistance would be greatly appreciated! Thank you.

**Comment by eginhard** [comment, marked as answer]
The error indicates that no samples for training were found. Without sharing the contents of your `recipes/multilingual/vits_tts/train_vits_tts.py` I can't help you further. Note that you'll need to modify that recipe for your data in case you haven't done so.

But you said that you just want to synthesise Thai speech from text. Is not not enough to just run that model? Why do you want to train it further?

Otherwise there's also a Thai TTS model from Fairseq included in Coqui that you could use directly: https://github.com/idiap/coqui-ai-TTS?tab=readme-ov-file#example-text-to-speech-using-fairseq-models-in-1100-languages-

**Comment by fws1** [comment]
Great, thank you for the reply. I do not need to train further, I misunderstood the steps involved. And, if there is an existing Fairseq Thai model then that's what I'm am looking for. I will definitely check it out.
